linode/manager
# feat: [M3-8953] - Object Storage Gen2 Updates and Improvements

## Changes  🔄
- `AccessKey`: Updated the "Regions/S3 Hostnames" view to align with the new design guidelines, improving how we display multiple region hostnames.
- `Non-Gen2 Capability Users`: Fixed the Properties tab visibility to respect Object Storage capabilities instead of relying on feature flags.
    - Added new e2e `bucket-details.spec.ts` to cover this
- `getAllBucketsFromEndpoints`: Fixed issue where duplicate buckets were being displayed for regions with more than one endpoint
- 🐛 Removed CORS toggle at the object-level. This was a bug that was doing nothing.

## Target release date 🗓️
12/10

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <img width="998" alt="Screenshot 2024-12-02 at 4 37 04 PM" src="https://github.com/user-attachments/assets/54a7d55f-b0d1-40f1-ae45-47c3bcd04df8"> | <img width="1003" alt="Screenshot 2024-12-03 at 9 34 05 AM" src="https://github.com/user-attachments/assets/1de15bb6-2642-41bb-9d73-2cd8c680225b"> |
| <img width="366" alt="Screenshot 2024-12-02 at 4 07 24 PM" src="https://github.com/user-attachments/assets/5ccabf1e-cae3-4b2c-966c-750bb85e922e"> | <img width="370" alt="Screenshot 2024-12-02 at 4 07 09 PM" src="https://github.com/user-attachments/assets/a5384be6-7906-4789-9114-819017d4e935"> |
| ![Screenshot 2024-12-04 at 12 05 09 PM](https://github.com/user-attachments/assets/6a22f705-4dc3-4015-83c7-ff6eeaccfc04) | ![Screenshot 2024-12-04 at 12 06 49 PM](https://github.com/user-attachments/assets/0f9abbe8-afde-4e65-a791-5660e2870a8f) |

## How to test 🧪

### Verification steps
1. Access Key
- Log in to dev account 
- Go to /object-storage/
- Go to /object-storage/access-keys and observe new "Regions/S3 Hostnames" format `| +x regions | Show All {copy icon}`

2. Non-Gen2 Capabilities
- Log into a dev test account `007` 
- Go to /object-storage/ and go to any bucket detail page
- Observe that you can no longer see the properties tab (Gen 2 feature)

3. Landing page and Create Access Key
- Confirm that duplicate buckets no longer appear in the landing page and in the drawer to create an access key (for Newark buckets)

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
